packages/notifi-react-card/lib/context/NotifiClientContext.tsx

```diff
@@ -47,19 +47,21 @@ export const NotifiClientContextProvider: React.FC<NotifiParams> = ({
   ...params
 }: React.PropsWithChildren<NotifiParams>) => {
   const client = useNotifiClient(params);
-
   const frontendClient = useMemo(() => {
     const configInput = getFrontendConfigInput(params);
-    const frontendClient = newFrontendClient(configInput);
-    frontendClient.initialize();
-    return frontendClient;
+    const updatedFrontendClient = newFrontendClient(configInput);
+    return updatedFrontendClient;
   }, [
     params.dappAddress,
     params.env,
     params.walletBlockchain,
     params.walletPublicKey,
   ]);
 
+  useEffect(() => {
+    frontendClient.initialize();
```

Comment:
race condition still exists for consumers of frontendClient, if consumer uses frontendClient before this initialize completes

Reply:
@juni-notifi Great idea, Thanks! Modified it accordingly.